packages/vscode-extension/src/plugins/network/strategies/DebuggerNetworkInspector.ts

```diff
@@ -134,15 +135,34 @@ export default class DebuggerNetworkInspector extends BaseNetworkInspector {
     this.disposables.push(...subscriptions, ...knownEventsSubscriptions);
   }
 
-  private setupBridgeAvailableListener(): void {
-    const bridgeAvailableSubscription = this.networkBridge.onEvent("bridgeAvailable", () => {
-      if (this.activationState === ActivationState.Pending) {
-        this.completeActivation();
-      }
-      bridgeAvailableSubscription.dispose();
-    });
+  private handleJsDebuggerConnected(): void {
+    if (this.activationState === ActivationState.Pending) {
+      this.completeActivation();
+    }
+
+    if (this.activationState === ActivationState.JsDebuggerDisconnected) {
+      // send Network.Enable method again to re-enable
+      // network-events reporting through js debugger
+      this.networkBridge.enableNetworkInspector();
+      this.activationState = ActivationState.Active;
+    }
+  }
+
+  private handleJsDebuggerDisconnected(): void {
+    if (this.activationState === ActivationState.Active) {
+      this.activationState = ActivationState.JsDebuggerDisconnected;
+    }
+  }
+
+  private setupDebuggerConnectionListeners(): void {
+    const debuggerConnectionSubscriptions: Disposable[] = [
+      this.networkBridge.onEvent("jsDebuggerConnected", () => this.handleJsDebuggerConnected()),
+      this.networkBridge.onEvent("jsDebuggerDisconnected", () =>
+        this.handleJsDebuggerDisconnected()
+      ),
+    ];
 
-    this.disposables.push(bridgeAvailableSubscription);
+    this.disposables.push(...debuggerConnectionSubscriptions);
```

Comment:
Just FYI, you could also write this as
```ts
this.disposables.push(
      this.networkBridge.onEvent("jsDebuggerConnected", () => this.handleJsDebuggerConnected()),
      this.networkBridge.onEvent("jsDebuggerDisconnected", () =>
        this.handleJsDebuggerDisconnected()
      ),
);
```
but if you feel the extra array variable makes it easier to read, feel free to leave it as is.